application/plugin_base/item_list.py

```diff
@@ -1647,6 +1665,19 @@ def destroy_providers(self):
 		for path, provider in self._providers.items():
 			provider.release_archive_handle()
 
+	def create_intermediate_archive_provider(self, path=None):
```

Comment:
Hi Mladen

I agree that it looks a bit weird - at least I should maybe try to remove
the recursion and make it a normal looping mechanism. As it is only called
from get_provider() then the lines could go there completely and your
concern is addressed?

A couple of aspects I have thought about:

1) when pressing CTRL LEFT/RIGHT the target panel is triggered and does
self.change_path(opposite_object.path) - only pulling the new path which
could be something like /home/or user/some.zip/subdr1/subdir2 with no
ZipProvider for /home/or user/some.zip attached to current (ie. target)
panel

2) I didn't see any code sharing of providers between panels - maybe
providers should be attached to to _parent window? the mine type handling
is already there as well as the list of available providers.

Looking at ZipProvider it only has one _zip_file member maybe there would
be a concurrency issue if the same archive (same path) is opened in both
panels? I don't know about other type of providers...

Let me what you think! Happy to rework it!

Cheers
Christian

On Mon, Feb 9, 2015 at 4:00 PM, Mladen Mijatov <notifications@github.com>
wrote:

> In application/plugin_base/item_list.py
> <https://github.com/MeanEYE/Sunflower/pull/65#discussion_r24338353>:
>
> > @@ -1647,6 +1665,19 @@ def destroy_providers(self):
> >  		for path, provider in self._providers.items():
> >  			provider.release_archive_handle()
> >
> > +	def create_intermediate_archive_provider(self, path=None):
>
> I understand the purpose, but am not really liking the way it works. It's
> a single purpose method for only specific use cases. There has to be a
> better way. Issue can be partially mitigated by getting provider from the
> opposite object, from which we are inheriting path anyway.
>
> That leaves issue of restoring path when restarting Sunflower, but I feel
> that should be a separate function.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/MeanEYE/Sunflower/pull/65/files#r24338353>.
>